docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,612 @@
+  val out = IO(new MyDecoupledOpt(false))
+  out :<>= in.waive(_.bits.get) // We can know to call .get because we can inspect in.bits.isEmpty
+}
+```
+
+This generates the following Verilog, where `ready` and `valid` are connected, and `bits` is ignored:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example6())
+```
+
+### Always ignore extra members (partial connection operator)
+
+The most unsafe connection is to connect only members that are present in both consumer and producer, and ignore all other members. This is unsafe because this connection will never error on any Chisel types.
+
+To do this, you can use `.waiveAll` and static cast to `Data`:
+
+```scala mdoc:silent
+class OnlyA extends Bundle {
+  val a = UInt(32.W)
+}
+class OnlyB extends Bundle {
+  val b = UInt(32.W)
+}
+class Example11 extends RawModule {
+  val in  = IO(Flipped(new OnlyA()))
+  val out = IO(new OnlyB())
+
+  out := DontCare
+
+  (out: Data).waiveAll :<>= (in: Data).waiveAll
+}
+```
+
+This generates the following Verilog, where nothing is connected:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example11())
+```
+
+## Techniques for connecting structurally inequivalent Chisel types
+
+`DataView` and `viewAsSupertype` creates a view of the component that has a different Chisel type. This means that a user can first create a `DataView` of the consumer or producer (or both) so that the Chisel types are structurally equivalent. This is useful when the difference between the consumer and producers aren't super nested, and also if they have rich Scala types which encode their structure. In general, `DataView` is the preferred mechanism to use (if you can) because it maintains the most about of Chisel information in the Scala type, but there are many instances where it doesn't work and thus one must fall back on `ConnectableData`.
+
+`ConnectableData` does not change the Chisel type, but instead changes the semantics of the operator to not error on the waived members if they are dangling or unassigned. This is useful for when differences between the consumer and producer do not show up in the Scala type system (e.g. present/missing fields of type `Option[Data]`, or anonymous `Record`s) or are deeply nested in a bundle that is especially onerous to create a `DataView`.
+
+Static casts (e.g. `(x: T)`) allows connecting components that have different Scala types, but leaves the Chisel type unchanged. Use this to force a connection to occur, even if the Scala types are different. One may wonder why the operators require identical Scala types in the first place, if they can easily be bypassed. The reason is to encourage users to use the Scala type system to encode Chisel information as it can make their code more robust; however, we don't want to be draconian about it because there are times when we want to enable the user to "just connect the darn thing".
```

Comment:
```suggestion
Static casts (e.g. `(x: T)`) allows connecting components that have different Scala types, but leaves the Chisel type unchanged. Use this to force a connection to occur, even if the Scala types are different.

 > One may wonder why the operators require identical Scala types in the first place, if they can easily be bypassed. The reason is to encourage users to use the Scala type system to encode Chisel information as it can make their code more robust; however, we don't want to be draconian about it because there are times when we want to enable the user to "just connect the darn thing".
```